src/CVRF/cvrf_priv.c

```diff
@@ -0,0 +1,1518 @@
+
+	return ret;
+}
+
+void cvrf_product_name_free(struct cvrf_product_name *full_name) {
+	if (full_name == NULL)
+		return;
+
+	oscap_free(full_name->cpe);
+	oscap_free(full_name->product_id);
+	oscap_free(full_name);
+}
+
+
+/***************************************************************************
+ ***************************************************************************
+ * Vulnerability offshoot of main CVRF model
+ *
+ */
+struct cvrf_vulnerability {
+	int ordinal;
+	char *vulnerability_title;
+	char *vulnerability_id;
+	char *discovery_date;
+	char *release_date;
+	char *cve_id;
+	char *cwe_id;
+
+	//Product Status
+	struct oscap_list *cvrf_product_statuses;
+	//Remediations
+	struct oscap_list *remediations;
+
+};
+OSCAP_ACCESSOR_STRING(cvrf_vulnerability, vulnerability_title)
+OSCAP_ACCESSOR_STRING(cvrf_vulnerability, vulnerability_id)
+OSCAP_ACCESSOR_STRING(cvrf_vulnerability, discovery_date)
+OSCAP_ACCESSOR_STRING(cvrf_vulnerability, release_date)
+OSCAP_ACCESSOR_STRING(cvrf_vulnerability, cve_id)
+OSCAP_ACCESSOR_STRING(cvrf_vulnerability, cwe_id)
+OSCAP_IGETINS_GEN(cvrf_product_status, cvrf_vulnerability, cvrf_product_statuses, cvrf_product_status)
+OSCAP_ITERATOR_REMOVE_F(cvrf_product_status)
+OSCAP_IGETINS_GEN(cvrf_remediation, cvrf_vulnerability, remediations, remediation)
+OSCAP_ITERATOR_REMOVE_F(cvrf_remediation)
+
+int cvrf_vulnerability_get_remediation_count(struct cvrf_vulnerability *vuln) {
+	return oscap_list_get_itemcount(vuln->remediations);
+}
+int cvrf_vulnerablity_get_ordinal(struct cvrf_vulnerability *vuln) {
+	return vuln->ordinal;
+}
+
+struct cvrf_vulnerability *cvrf_vulnerability_new() {
+
+	struct cvrf_vulnerability *ret;
+
+	ret = oscap_alloc(sizeof(struct cvrf_vulnerability));
+	if (ret == NULL)
+		return NULL;
+
+	ret->vulnerability_title = NULL;
+	ret->vulnerability_id = NULL;
+	ret->discovery_date = NULL;
+	ret->release_date = NULL;
+	ret->cve_id = NULL;
+	ret->cwe_id = NULL;
+	ret->cvrf_product_statuses = oscap_list_new();
+	ret->remediations = oscap_list_new();
+
+	return ret;
+}
+
+void cvrf_vulnerability_free(struct cvrf_vulnerability *vulnerability) {
+
+	if (vulnerability == NULL)
+		return;
+
+	oscap_free(vulnerability->vulnerability_title);
+	oscap_free(vulnerability->vulnerability_id);
+	oscap_free(vulnerability->discovery_date);
+	oscap_free(vulnerability->release_date);
+	oscap_free(vulnerability->cve_id);
+	oscap_free(vulnerability->cwe_id);
+	oscap_list_free(vulnerability->cvrf_product_statuses, (oscap_destruct_func) cvrf_product_status_free);
+	oscap_list_free(vulnerability->remediations, (oscap_destruct_func) cvrf_remediation_free);
+	oscap_free(vulnerability);
+}
+
+
+/***************************************************************************
+ * CVRF Remediation
+ */
+struct cvrf_remediation {
+	cvrf_remediation_type_t remed_type;
+	char *remed_date;
+	char *remed_description;
+	char *remed_URL;
+	char *remed_entitlement;
+	char *remed_product_id;
+	char *remed_group_id;
```

Comment:
drop the member prefixes here. They are all members of `cvrf_remediation` so the prefix is unnecessary.